src/ComplexityVisitor.js

```diff
@@ -1,8 +1,46 @@
 import { GraphQLList, GraphQLNonNull, GraphQLObjectType } from 'graphql';
+import { print } from 'graphql/language/printer';
 import * as IntrospectionTypes from 'graphql/type/introspection';
 
 import CostCalculator from './CostCalculator';
 
+function isSameArguments(arguments1, arguments2) {
+  if (arguments1.length !== arguments2.length) {
+    return false;
+  }
+
+  return arguments1.every((argument1) => {
+    const argument2 = arguments2.find(
+      ({ name }) => name.value === argument1.name.value,
+    );
+
+    if (!argument2) {
+      return false;
+    }
+
+    return print(argument1.value) === print(argument2.value);
+  });
+}
+
+function isSameSelection(selection1, selection2) {
+  return (
+    selection1.kind === selection2.kind &&
+    selection1.name?.value === selection2.name?.value &&
+    isSameArguments(selection1.arguments || [], selection2.arguments || [])
+  );
+}
```

Comment:
should probably do something with directives?

Reply:
directives (at least per relay) aren't supposed to affect execution in that way, so they shouldn't necessarily affect costs